pkg/utils/transform/transform_test.go

```diff
@@ -398,3 +398,73 @@ func assertTaskImage(t *testing.T, resources []unstructured.Unstructured, name s
 		}
 	}
 }
+
+func TestTransformManifest_InjectNamespaceRoleBindingSubjects(t *testing.T) {
+	resourceWithAnnotation := "testdata/inject-ns-rolebinding.yaml"
+	manifest, err := mf.NewManifest(resourceWithAnnotation, true, nil)
+	assertNoEror(t, err)
+	tf := InjectNamespaceRoleBindingSubjects("target")
+	err = manifest.Transform(tf)
+	assertNoEror(t, err)
+	assertRBSubjectNamespace(t, manifest.Resources[0], "target")
+}
+
+func TestTransformManifest_InjectNamespaceRoleBindingSubjects_NoNs(t *testing.T) {
+	resourceWithAnnotation := "testdata/inject-ns-rolebinding-no-ns.yaml"
+	manifest, err := mf.NewManifest(resourceWithAnnotation, true, nil)
+	assertNoEror(t, err)
+	tf := InjectNamespaceRoleBindingSubjects("target")
+	err = manifest.Transform(tf)
+	assertNoEror(t, err)
+	assertRBSubjectNamespace(t, manifest.Resources[0], "")
+}
+
+func TestTransformManifest_InjectNamespaceCRDWebhookClientConf(t *testing.T) {
+	resourceWithAnnotation := "testdata/inject-ns-crd-webhookclientconf.yaml"
+	manifest, err := mf.NewManifest(resourceWithAnnotation, true, nil)
+	assertNoEror(t, err)
+	tf := InjectNamespaceCRDWebhookClientConfig("target")
+	err = manifest.Transform(tf)
+	assertNoEror(t, err)
+	assertCRDWebhookClientConfNS(t, manifest.Resources[0], "target")
+}
+
+func TestTransformManifest_InjectNamespaceCRDWebhookClientConf_NoNs(t *testing.T) {
+	resourceWithAnnotation := "testdata/inject-ns-crd-webhookclientconf-no-ns.yaml"
+	manifest, err := mf.NewManifest(resourceWithAnnotation, true, nil)
+	assertNoEror(t, err)
+	tf := InjectNamespaceCRDWebhookClientConfig("target")
+	err = manifest.Transform(tf)
+	assertNoEror(t, err)
+	assertCRDWebhookClientConfNS(t, manifest.Resources[0], "target")
```

Comment:
@pradeepitm12  the last parameter should be `""`. 
the tests are passing because we have the `if _, ok := m["namespace"]; ok {` check in the assertions.

it will be better if you could split both assertions (crd and rolebinding) into 2 (total 4)
- assertRBSubjectNamespace
- assertRBSubjectNoNamespace
- assertCRDWebhookClientConfNS
- assertCRDWebhookClientConfNoNS

it will be more readable. It took me a minute and a try with golang repl to understand why `assertCRDWebhookClientConfNS(t, manifest.Resources[0], "target")` passed. :) 

nit: let us stick to `Namespace` or `NS` (assertRBSubjectNamespace and assertCRDWebhookClientConfNS)

Reply:
i appreciate you adding the negative tests as well :+1:

Reply:
I actually wrote the positive test and then added the negative one while adding the negative I added `if _, ok := m["namespace"]; ok {`.
But it actually makes sense to split both the assertions.
Thank you.